### PR TITLE
AP photogram industrial 2

### PR DESCRIPTION
**Pull request by APatel-AI**
- Generated and migrated Comments, Follow Requests, and Likes tables using scaffold
- updated route connections

### CODE REVIEW

**Comment by thierrychau** [comment]
For future reference: please only tag for a review once it ready. I see you are still committing to this, so not sure when it's ready for review.

Drop a comment here when it is.

**Comment by APatel-AI** [comment]
Ready for review

**Comment by MayaS1111** [comment]
Looks good to me, let's see what Thierry is going to say

**Review by MayaS1111 — APPROVED** [review]
Looks good to me, let's see what Thierry is going to say

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#
+#  fk_rails_...  (fan_id => users.id)
+#  fk_rails_...  (photo_id => photos.id)
+#
+class Like < ApplicationRecord
+  belongs_to :fan, class_name: "User", counter_cache: true
+  belongs_to :photo, counter_cache: true
+
+
+  validates :user_id, uniqueness: { scope: :photo_id, message: "has already liked this photo" }
```
Comment (thierrychau) [inline app/models/like.rb]: @APatel-AI you have a validation on a field named `user_id`, but in your like model you do not have this field. Can you update this?

File: app/models/user.rb [hunk 2]
```diff
@@ -27,5 +28,26 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
 
-  has_many :own_photos, class_name: "Photo", foreign_key: "owner_id"
+
+  has_many :sent_follow_requests, foreign_key: :sender_id, class_name: "FollowRequest"
```
Comment (thierrychau) [inline app/models/user.rb]: Can you add a `dependent: :destroy` to handle orphan records?

**Review by thierrychau — CHANGES_REQUESTED** [review]
Hi Arpan, great job so far. I have a few minor comments, please see below. Thanks!

**Comment by APatel-AI** [comment]
added dependent: :destroy to handle orphan records and updated the validation for like model

**Review by dylansarikas — APPROVED** [review]
Looks good Arpan!!  Neat and easy to follow.

**Review by thierrychau — APPROVED** [review]
Great job, good to merge. Thanks for the edits.

**Pull request merged**